src/coreclr/jit/gentree.cpp

```diff
@@ -2212,22 +2217,59 @@ void GenTreeCall::AddGDVCandidateInfo(Compiler* comp, InlineCandidateInfo* candi
 
     if (gtInlineInfoCount == 0)
     {
+        // Most calls are monomorphic, so we don't need to allocate a vector
         gtInlineCandidateInfo = candidateInfo;
     }
     else if (gtInlineInfoCount == 1)
     {
-        gtInlineCandidateInfo =
-            new (comp, CMK_Inlining) InlineCandidateInfo[MAX_GDV_TYPE_CHECKS]{*gtInlineCandidateInfo, *candidateInfo};
+        // Upgrade gtInlineCandidateInfo to gtInlineCandidateInfoList (vector)
+        CompAllocator       allocator          = comp->getAllocator(CMK_Inlining);
+        InlineCandidateInfo firstCandidateInfo = *gtInlineCandidateInfo;
```

Comment:
If you stick with this vector of structs then you can at least not deref here and save one copy; as is, this moderately large struct gets copied twice, first to this local and then later to the vector.